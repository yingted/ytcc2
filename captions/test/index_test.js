'use strict';
let {
  decodeTimeSpaceFast,
  decodeTimeSpace,
} = require('../index.js');

describe('decodeTimeSpaceFast matches decodeTimeSpace', () => {

  let testMatches = function testMatches(input) {
    test(input, () => {
      expect(decodeTimeSpaceFast(input)).toStrictEqual(decodeTimeSpace(input));
      expect(decodeTimeSpaceFast(input)).toMatchSnapshot();
    });
  };

  // non-timestamps
  testMatches('');
  testMatches('a');
  testMatches('1');
  testMatches('1:');
  testMatches('1:.');
  testMatches('1:. ');

  // timestamps
  testMatches('1:2');
  testMatches('1:2 ');
  testMatches('1:2a');
  testMatches('1:2.');
  testMatches('1:2.a');
  testMatches('0:1:02');
  testMatches('1:02');
  testMatches('1:02 ');
  testMatches('1:02 a');
  testMatches('1:02.');
  testMatches('1:02.a');
  testMatches('1:02.3');
  testMatches('1:.3');
  testMatches('1:.3 ');
  testMatches('1:.3 a');
});
