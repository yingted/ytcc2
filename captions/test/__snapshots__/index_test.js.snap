// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decodeTimeSpaceFast matches decodeTimeSpace  1`] = `null`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 0:1:02 1`] = `
Object {
  "offset": 6,
  "time": 62,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1 1`] = `null`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1: 1`] = `null`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:.  1`] = `null`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:. 1`] = `null`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:.3  1`] = `
Object {
  "offset": 5,
  "time": 60.3,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:.3 1`] = `
Object {
  "offset": 4,
  "time": 60.3,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:.3 a 1`] = `
Object {
  "offset": 5,
  "time": 60.3,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:02  1`] = `
Object {
  "offset": 5,
  "time": 62,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:02 1`] = `null`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:02 a 1`] = `
Object {
  "offset": 5,
  "time": 62,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:02. 1`] = `
Object {
  "offset": 5,
  "time": 62,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:02.3 1`] = `
Object {
  "offset": 6,
  "time": 62.3,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:02.a 1`] = `
Object {
  "offset": 5,
  "time": 62,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:2  1`] = `
Object {
  "offset": 4,
  "time": 62,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:2 1`] = `null`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:2. 1`] = `
Object {
  "offset": 4,
  "time": 62,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:2.a 1`] = `
Object {
  "offset": 4,
  "time": 62,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace 1:2a 1`] = `
Object {
  "offset": 3,
  "time": 62,
}
`;

exports[`decodeTimeSpaceFast matches decodeTimeSpace a 1`] = `null`;
